features/netsocket/SocketAddress.cpp

```diff
@@ -180,7 +183,8 @@ const char *SocketAddress::get_ip_address() const
         return NULL;
     }
 
-    if (!_ip_address[0]) {
+    if (!_ip_address) {
+        _ip_address = new char[NSAPI_IP_SIZE];
```

Comment:
I guess you need an "else" here now, as this will leave the address undefined, not "", if somehow version is not v4 or v6.

Maybe you could actually put the "new" in the ifs? Would allow you to have a smaller buffer for IPv4.

Reply:
(Another option would be to convert into stack temporary, then alloc exactly the right size. But a bit wary of using more stack).

Reply:
Couple of lines up, it is already checked that address type is not `NSAPI_UNSPEC` if so, it returns NULL before allocating the space.

Yes, using a stack space, then copying the string by `strdup()` would use less heap, but do we want to optimise for this? For me, it sounds like should be used only once per application.